app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -0,0 +1,195 @@
+package com.kickstarter.ui.activities
+
+import android.content.DialogInterface
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.WindowManager
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.jakewharton.rxbinding.view.RxView
+import com.jakewharton.rxbinding.widget.RxTextView
+import com.kickstarter.R
+import com.kickstarter.databinding.CommentsLayoutBinding
+import com.kickstarter.libs.ActivityRequestCodes
+import com.kickstarter.libs.BaseActivity
+import com.kickstarter.libs.RecyclerViewPaginator
+import com.kickstarter.libs.SwipeRefresher
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.ObjectUtils
+import com.kickstarter.libs.utils.TransitionUtils
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.models.Project
+import com.kickstarter.ui.IntentKey
+import com.kickstarter.ui.adapters.CommentsAdapter
+import com.kickstarter.ui.data.LoginReason
+import com.kickstarter.ui.viewholders.EmptyCommentsViewHolder
+import com.kickstarter.ui.viewholders.ProjectContextViewHolder
+import com.kickstarter.viewmodels.CommentsViewModel
+import com.trello.rxlifecycle.ActivityEvent
+import rx.Observable
+import rx.android.schedulers.AndroidSchedulers
+import rx.subjects.PublishSubject
+
+@RequiresActivityViewModel(CommentsViewModel.ViewModel::class)
+class CommentsActivity : BaseActivity<CommentsViewModel.ViewModel>(), CommentsAdapter.Delegate {
+    private var adapter: CommentsAdapter? = null
```

Comment:
these 3 optionals can be changed to latinits
`  private lateinit var adapter: CommentsAdapter
    private lateinit var recyclerViewPaginator: RecyclerViewPaginator
    private lateinit var swipeRefresher: SwipeRefresher`